test/test-data-unicode.json

```diff
@@ -985,5 +985,56 @@
       4
     ],
     "raw": "[^}]"
+  },
+  "[\\b-e]": {
+    "type": "characterClass",
+    "body": [
+      {
+        "type": "characterClassRange",
+        "min": {
+          "type": "value",
+          "kind": "singleEscape",
+          "codePoint": 8,
+          "range": [
+            1,
+            3
+          ],
+          "raw": "\\b"
+        },
+        "max": {
+          "type": "value",
+          "kind": "symbol",
+          "codePoint": 101,
+          "range": [
+            4,
+            5
+          ],
+          "raw": "e"
+        },
+        "range": [
+          1,
+          5
+        ],
+        "raw": "\\b-e"
+      }
+    ],
+    "negative": false,
+    "range": [
+      0,
+      6
+    ],
+    "raw": "[\\b-e]"
+  },
+  "[e-\\b]": {
+    "type": "error",
+    "name": "SyntaxError",
+    "message": "invalid range in character class at position 1: e-\\b\n    [e-\\b]\n     ^",
+    "input": "[e-\\b]"
+  },
```

Comment:
Maybe you can also add `[\0-\b]` as a test, which is a valid regexp where `\b` appears in the second position.

Reply:
Added the tests with and without unicode flag. They are both looking good.